tests/integration/test_issue_590_regression.py

```
@@ -0,0 +1,100 @@
+@pytest.mark.asyncio
+async def test_failed_pipeline_run_span_finalization_regression(temp_db_path):
+    """
+    Regression test for Issue #590 - Failed run case.
+    Ensures that even if a pipeline fails, the root span is marked as 'failed' (or completed/processed)
+    and has a non-null end_time in the database.
+    """
+    backend = SQLiteBackend(temp_db_path)
+    try:
+
+        class FailingAgent:
+            async def run(self, input_data):
```

Ruff (0.14.13)

Unused method argument: `input_data`

(ARG002)